modules/statistics/php/statistics_site.class.inc

```diff
@@ -55,6 +55,20 @@ class Statistics_Site extends \NDB_Menu
                 'data_entry',
                 intval($_REQUEST['CenterID'])
             );
+        } else {
+
+            // If no CenterID is passed in the request, check if the user has the
+            // data_entry permission in all the sites/centers it have access to.
+            // TODO: There are no means of set permissions per site
+            // for a given user right now: (e.g.) The user X can have
+            // the permission data_entry on site Y but not on site Z.
+
+            $hasCenterPermission = true;
+            foreach ($user->getStudySites() as $key => $value) {
+                if (!$user->hasCenterPermission('data_entry', intval($key))) {
+                    $hasCenterPermission = false;
+                }
```

Comment:
I think it's better here to use the `getCenterIDs()` from the user class.

Also, I'm not sure this does what you intend it to do. It's true currently you can not set per-site permissions so I'm not sure why we are using `hasCenterPermission()` but in the even this feature gets added. A user in this case will be denied access to the entire page if they dont have `data_entry` on a single one of their sites. 

I think a better way of doing this check is to return true if the user has `data_entry` on any of their sites, not all of their sites...

Reply:
Hi @ridz1208,

"Also, I'm not sure this does what you intend it to do. It's true currently you can not set per-site permissions so I'm not sure why we are using hasCenterPermission() but in the even this feature gets added. A user in this case will be denied access to the entire page if they dont have data_entry on a single one of their sites" **... hahaha... yes, in what I was thinking...**

I my opinion the long term robust solution will be to not allow access if the centerIDs aren't specified during the GET query. This will take a bit more of coding and a slightly modification on the link that calls this page. We could pass an array of centerIDs instead.

![CenterIDs](https://user-images.githubusercontent.com/37309344/69159657-47042100-0ab6-11ea-9719-9126b376d3dc.png)

For the **short term solution** I would propose the following:

Let's say we leave the` hasCenterPermission()` function with the name it has now as **the semantic** (I didn't developed this one but I think) is the one we're searching for: **to check if a user have permissions for a given site.** Even if its not the way it is implemented right now... we could embed the logic inside after, when the permission framework ready. Also not sure across how many pages this function is used as it seems to be a core one.

During the filtering, in `_checkCriteria` we could incorporate the logic to only get the sites in which the user have the **'data_entry'** permission using the function `hasCenterPermission()`, **which for the moment will do not what is intended but will don't break it neither.**  

Then, once we implemented the correct logic inside `hasCenterPermission()` the system should be compatible and effectively filtering.

Regarding `getCentersIDs()` and `getStudySites()` the main difference I found so far is that `getStudySites()` is checking if the center is flagged as a study one? ... I'm not familiar with this two functions or the business logic they encapsulated... any light in this sense is more than welcome. 

Thanks,
Rolando (CHU Sainte-Justine)

Reply:
assuming we keep `hasCenterPermission`, there is no reason to block access to a user who has `data_entry` on only one of their `n` sites...

I think the only check here should be, does the user have data_entry on any of their sites ? if yes, allow access and filter the data below (already in this PR); if no, then block access.

I'm reading my proposition from above

> In this particular case, what I would do is remove the if statement on line 53 simply do a check if the user has data_entry allow access to the page, otherwise block access. the filtering below will make sure user does not have access to the incorrect data